src/org/traccar/protocol/ArnaviProtocolDecoder.java

```diff
@@ -79,12 +79,14 @@ protected Object decode(
         position.set(Position.KEY_BATTERY, parser.nextInt() * 0.01);
         position.set(Position.KEY_IGNITION, parser.nextInt() == 1);
         position.set(Position.KEY_INPUT, parser.nextInt());
-        position.set(Position.KEY_SATELLITES, parser.nextInt());
+
+        int satellites = parser.nextInt();
+        position.set(Position.KEY_SATELLITES, satellites);
 
         DateBuilder dateBuilder = new DateBuilder()
                 .setTime(parser.nextInt(), parser.nextInt(), parser.nextInt());
 
-        position.setValid(true);
+        position.setValid(satellites >= 3);
```

Comment:
I will rework the pull request to follow your comment.  I need to know about the other methods used to Position.setValid().  Please review the code base for all usage of Position.setValid() and I will follow the method that you prefer.